sensing/autoware_pointcloud_preprocessor/src/concatenate_data/concatenate_and_time_sync_node.cpp

```diff
@@ -0,0 +1,500 @@
+    RCLCPP_ERROR(
+      get_logger(), "The number of topics does not match the number of timestamp offsets");
+    return;
+  }
+  if (params_.lidar_timestamp_noise_window.size() != params_.input_topics.size()) {
+    RCLCPP_ERROR(
+      get_logger(), "The number of topics does not match the number of timestamp noise windwo");
+    return;
+  }
+
+  for (size_t i = 0; i < params_.input_topics.size(); i++) {
+    topic_to_offset_map_[params_.input_topics[i]] = params_.lidar_timestamp_offsets[i];
+    topic_to_noise_window_map_[params_.input_topics[i]] = params_.lidar_timestamp_noise_window[i];
+  }
+
+  // Publishers
+  concatenate_cloud_publisher_ = this->create_publisher<sensor_msgs::msg::PointCloud2>(
+    "output", rclcpp::SensorDataQoS().keep_last(params_.maximum_queue_size));
+
+  // Transformed Raw PointCloud2 Publisher to publish the transformed pointcloud
+  if (params_.publish_synchronized_pointcloud) {
+    for (auto & topic : params_.input_topics) {
+      std::string new_topic =
+        replaceSyncTopicNamePostfix(topic, params_.synchronized_pointcloud_postfix);
+      auto publisher = this->create_publisher<sensor_msgs::msg::PointCloud2>(
+        new_topic, rclcpp::SensorDataQoS().keep_last(params_.maximum_queue_size));
+      topic_to_transformed_cloud_publisher_map_.insert({topic, publisher});
+    }
+  }
+
+  // Subscribers
+  if (params_.is_motion_compensated) {
+    if (params_.input_twist_topic_type == "twist") {
+      twist_sub_ = this->create_subscription<geometry_msgs::msg::TwistWithCovarianceStamped>(
+        "~/input/twist", rclcpp::QoS{100},
+        std::bind(
+          &PointCloudConcatenateDataSynchronizerComponent::twist_callback, this,
+          std::placeholders::_1));
+    } else if (params_.input_twist_topic_type == "odom") {
+      odom_sub_ = this->create_subscription<nav_msgs::msg::Odometry>(
+        "~/input/odom", rclcpp::QoS{100},
+        std::bind(
+          &PointCloudConcatenateDataSynchronizerComponent::odom_callback, this,
+          std::placeholders::_1));
+    } else {
+      RCLCPP_ERROR_STREAM(
+        get_logger(), "input_twist_topic_type is invalid: " << params_.input_twist_topic_type);
+      throw std::runtime_error(
+        "input_twist_topic_type is invalid: " + params_.input_twist_topic_type);
+    }
+  }
+
+  pointcloud_subs.resize(params_.input_topics.size());
+  for (size_t topic_id = 0; topic_id < params_.input_topics.size(); ++topic_id) {
+    std::function<void(const sensor_msgs::msg::PointCloud2::SharedPtr msg)> callback = std::bind(
+      &PointCloudConcatenateDataSynchronizerComponent::cloud_callback, this, std::placeholders::_1,
+      params_.input_topics[topic_id]);
+
+    pointcloud_subs[topic_id].reset();
+    pointcloud_subs[topic_id] = this->create_subscription<sensor_msgs::msg::PointCloud2>(
+      params_.input_topics[topic_id], rclcpp::SensorDataQoS().keep_last(params_.maximum_queue_size),
+      callback);
+  }
```

Comment:
Change this to a range-based for loop and do not use indices for robustness/maintainability:
```cpp
for (const std::string & topic : params_input_topics) {
  auto sub = create_subscription...;
  pointcloud_subs.push_back(sub);
}
```

Reply:
Thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/b344427281484082c80fd8c5b3993754652dd91d